tests/pipeline/test_pipeline.py

```diff
@@ -480,3 +480,46 @@ def test_validate_unique_step_names(self):
                 ],
                 catch_exceptions=False,
             )
+
+    def test_validate_loops_on_pipeline(self):
+        with pytest.raises(ValueError, match="Pipeline contains cycles."):
+            runner.invoke(
+                app,
+                [
+                    "generate",
+                    "--pipeline-base-dir",
+                    str(PIPELINE_BASE_DIR_PATH),
+                    str(RESOURCE_PATH / "pipeline-with-loop/pipeline.yaml"),
+                    "--defaults",
+                    str(RESOURCE_PATH / "pipeline-with-loop"),
+                ],
+                catch_exceptions=False,
+            )
+
+    def test_validate_simple_graph(self):
+        pipeline = kpops.generate(
+            RESOURCE_PATH / "pipelines-with-graphs/simple-pipeline/pipeline.yaml",
+            pipeline_base_dir=PIPELINE_BASE_DIR_PATH,
+            defaults=RESOURCE_PATH / "pipelines-with-graphs" / "simple-pipeline",
+        )
+        assert len(pipeline.components) == 2
+        assert len(pipeline.components.graph.nodes) == 3
+        assert len(pipeline.components.graph.edges) == 2
+        node_components = list(
+            filter(lambda x: "component" in x, pipeline.components.graph.nodes)
+        )
+        assert len(pipeline.components) == len(node_components)
+
+    def test_validate_topic_and_component_same_name(self):
+        pipeline = kpops.generate(
+            RESOURCE_PATH
+            / "pipelines-with-graphs/same-topic-and-component-name/pipeline.yaml",
+            pipeline_base_dir=PIPELINE_BASE_DIR_PATH,
+            defaults=RESOURCE_PATH
+            / "pipelines-with-graphs"
+            / "same-topic-and-component-name",
+        )
+        nodes = list(pipeline.components.graph.nodes)
+        edges = list(pipeline.components.graph.edges)
+        assert nodes[2] == f"component-{nodes[3]}"
+        assert (nodes[2], nodes[3]) in edges
```

Comment:
please make it more clear which node is which
I am assuming `nodes[2]` is the component (streams-app?) and `nodes[3]` is the topic in this case. For example, you could unpack the `nodes` list